tfutils/base.py

```diff
@@ -900,71 +908,134 @@ def train_from_params(save_params,
                   'learning_rate_params': learning_rate_params,
                   'optimizer_params': optimizer_params,
                   'validation_params': validation_params,
-                  'thres_loss': thres_loss,
-                  'num_steps': num_steps,
                   'log_device_placement': log_device_placement}
         dbinterface = DBInterface(sess=sess, global_step=global_step, params=params,
                                   save_params=save_params, load_params=load_params)
         dbinterface.initialize()
-        train(sess,
-              queues,
-              dbinterface,
-              train_targets=train_targets,
-              global_step=global_step,
-              num_steps=num_steps,
-              valid_targets=valid_targets_dict,
-              thres_loss=thres_loss)
+        return train(sess,
+                     queues,
+                     dbinterface,
+                     train_targets=train_targets,
+                     global_step=global_step,
+                     num_steps=train_params['num_steps'],
+                     thres_loss=train_params['thres_loss'],
+                     validation_targets=valid_targets_dict)
 
 
 def get_valid_targets_dict(validation_params,
```

Comment:
I'm not sure which functions are "public" yet.  But if you want to go and make some decisions like this after I merge the PR, fine with me.